src/components/ScholarshipForm.jsx

```diff
@@ -41,29 +43,28 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-function ScholarshipForm({ scholarship, submitFn, onSubmitError }) {
+function ScholarshipForm({ scholarship, onSubmit }) {
   const classes = useStyles();
   const [activeStep, setActiveStep] = useState(0);
-  const [noReqsChecked, setNoReqsChecked] = useState(false);
+  const [submissionError, setSubmissionError] = useState(null);
 
   const formik = useFormik({
     initialValues: scholarship.data,
     validationSchema,
     validateOnChange: false,
-    onSubmit: (values, { setSubmitting, resetForm }) => {
+    onSubmit: (values, { setSubmitting }) => {
       setSubmitting(true);
       scholarship.data = { ...values };
       scholarship
         .save()
-        .then(submitFn)
-        .then(() => setActiveStep(0))
-        .then(() => setNoReqsChecked(false))
-        .then(resetForm)
-        .catch(onSubmitError)
+        .then(onSubmit)
+        .catch(setSubmissionError)
         .finally(() => setSubmitting(false));
     },
   });
 
+  const noReqsChecked = JSON.stringify(formik.values.requirements) === '{}';
```

Comment:
Nice! 
I was curious why `JSON.stringify` was needed but I am guessing so we can make this check easier.

Reply:
Yepp. Added a comment to help explain it since it's not intuitive at first glance.